src/poll.rs

```diff
@@ -145,3 +270,339 @@ pub fn selector(poll: &Poll) -> &sys::Selector {
 pub fn selector_mut(poll: &mut Poll) -> &mut sys::Selector {
     &mut poll.selector
 }
+
+/*
+ *
+ * ===== Registration =====
+ *
+ */
+
+impl Registration {
+    pub fn new(poll: &Poll, token: Token, interest: EventSet, opts: PollOpt) -> Registration {
+        let queue = poll.readiness_queue.clone();
+        let node = queue.new_readiness_node(token, interest, opts);
+
+        Registration {
+            node: node,
+            queue: queue,
+        }
+    }
+
+    pub fn update(&self, poll: &Poll, interest: EventSet, opts: PollOpt) -> io::Result<()> {
```

Comment:
Since this `Registration` is already inherently tied to the original `Poll`, is there a benefit from taking it as an extra parameter here? It'd perhaps be nicer to not have to pass it?